jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/KafkaProxySampleTest.java

```diff
@@ -0,0 +1,188 @@
+        kafka.start();
+        kafkaFixture.createTopic(kafka);
+        proxyBootstrap = kafkaProxy.getContainerIpAddress()+":"+kafkaProxy.getProxyPort();
+        
+        System.setProperty(BOOTSTRAP_SERVERS_CONFIG, proxyBootstrap);
+        System.setProperty(CLIENT_ID_CONFIG, "test_jbpm");
+        System.setProperty(KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        System.setProperty(VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        
+        deploymentId = kafkaFixture.setup(deploymentService, strategy);
+        
+        countDownLatchEventListener.setVariable(KAFKA_RESULT);
+    }
+
+    @After
+    public void cleanup() {
+        kafka.stop();
+        toxiproxy.stop();
+        kafkaFixture.cleanup(deploymentService);
+    }
+
+    @Test(timeout = 240000)
+    public void testKafkaWIHNoConnection() throws Exception {
+
+        countDownLatchEventListener.configureNode(KAFKA_PROCESS_ID, "TaskErrorAfterKafkaMessageSent", 2);
+        
+        kafkaProxy.setConnectionCut(true);
+        
+        //Kafka WIH will try during  publish config max.block.ms -60 seconds by default- to get connected to Kafka
+        //TimeoutException: Topic mytopic not present in metadata after 60000 ms.
+        Long processInstanceId = processService.startProcess(deploymentId,
+                                                             KAFKA_PROCESS_ID,
+                                                             kafkaFixture.getProcessVariables());
+
+        assertTrue(processInstanceId > 0);
+
+        //Countdown decrements the count of the latch twice: 
+        //TaskErrorAfterKafkaMessageSent node and before process ends
+        countDownLatchEventListener.getCountDown().await();
+
+        assertEquals("failure", (String)countDownLatchEventListener.getResult());
+    }
+
+    
+    @Test(timeout = 60000)
+    public void testKafkaWIHReconnect() throws Exception {
+
+        countDownLatchEventListener.configure(KAFKA_PROCESS_ID, 1);
+        
+        kafkaProxy.setConnectionCut(true);
+        
+        reconnectProxyLater(10);
+        
+        Long processInstanceId = processService.startProcess(deploymentId,
+                                                             KAFKA_PROCESS_ID,
+                                                             kafkaFixture.getProcessVariables());
+
+        assertTrue(processInstanceId > 0);
+
+        kafkaFixture.assertConsumerMessages(proxyBootstrap);
+
+        //Countdown decrements the count of the latch before process ends
+        countDownLatchEventListener.getCountDown().await();
+        
+        assertEquals("success", countDownLatchEventListener.getResult());
+    }
+
+    private void reconnectProxyLater(int reconnectTime) {
+        new Thread(() -> {
+            CountDownLatch lock = new CountDownLatch(1);
+            try {
+                lock.await(reconnectTime, TimeUnit.SECONDS);
```

Comment:
Why not to use just sleep here as there is no way to call countDown on the latch anyway.

Reply:
right, this approach is just to avoid sleep and be more sonarcloud rules compliant

Reply:
OK, makes sense now.